### ISSUE TITLE
Targeted Survey UI Step 2: Choose your Audience

### ISSUE THREAD

**Issue by willdoran** [ushahidi/platform#2528]
### Overview
Allows user to specify the recipients of the SMS message.

Note: Design is currently revising based on User Test feedback, ensure that the Designs are up to date before completing this issue by checking with @Erioldoesdesign or @justinscherer                                               

Data validation needs to be defined for this step.

![Screen Shot 2018-02-27 at 14.22.33.png](https://images.zenhubusercontent.com/5a6a44264b5806bc2bc4546d/65ce2361-81be-45d6-9ea3-d44a74ef4827)

### Requirements
- [ ] User can enter a comma separate list of telephone numbers
- [ ] The field is required
- [ ] The user can only proceed to the next step when the field is filled

### Dependencies
- #2526

### Test script
- [ ] Phone numbers field is available and can be filled
- [ ] User can not proceed to next step if field is blank
- [ ] User can proceed to next step if field is filled

**Comment by justinscherer** [comment]
Is the field validation in a different issue?

**Comment by rowasc** [comment]
@justinscherer not yet. But we should figure it out (initially ) in each ticket and then we can divide them if it makes sense.
Q: What does the validation for the audience list look like? I don't see it in the prototype (unless it's the same way as other validations) 
Also, about validation rules: are we validating phone numbers or just the list format? if we are validating phone numbers, how do we show the error for each invalid number? 
cc @willdoran

**Comment by willdoran** [comment]
@justinscherer @rowasc There is no field validation at the moment. My understanding was that we are waiting on changes based on User Testing to determine if the numbers should be typed in or uploaded via CSV that will impact field validation for this. 

For all other inputs field validation will be the same as present field validation - though I think there are only text inputs on this screen, apart from this one, no?

**Comment by willdoran** [comment]
- Is there a limit on the number of numbers?

**Comment by willdoran** [comment]
For the moment, we are going to stick to text input box approach

**Comment by justinscherer** [comment]
Can we try to do inline validation on:
- field can't be empty (i.e. it's required)
- can't have more than X numbers (@willdoran created another issue for this)
- every number has 10 digits (assuming it's a global standard...)


When we find a problem, can we quote the number between commas that has a problem in the error. e.g. "Each number must have 10 digits and be separated by a comma. The following numbers are in an incorrect format: 12345678910..."

**Comment by willdoran** [comment]
@justinscherer I think we can do the first without issue.

For the second and third we can use a phone number validation library, we should use the same one as Tenfour
https://github.com/ushahidi/RollCall/blob/0dc838f7d0e4cac26dc2f01ceb7c8bcaf067f414/app/Validators/PhoneNumberValidator.php

**Comment by crcommons** [comment]
Adding checklist for testing: 
Step 1 and step 2.

Still needs work:
- functioning, mocked endpoint for country data
- Fix alert div styles to be blocks, rather than inline
- how to handle if user inputs ' ' or '   ',  etc... if at all? currently only handle totally empty string: ''

Testing checklist:
Step 1:
- [ ] Attempt to "Continue" to step 2 without filling in step 1; alerts should appear for both fields
- [ ] Fill in "Survey Name" with one character; warning about response being too short should appear
- [ ] Fill in data for Step 1 fields; alert messages should go away
- [ ] select "Continue", you should move to step 2 and Survey name and Survey Description should remain

Step 2
- [ ] Attempt to continue to step 3 without filling in step 2; alerts should appear for both fields
- [ ] select United States; alert should disappear 
- [ ] paste these numbers 3172956778, 2025468989; alert should disappear
- [ ] select continue; you should move to step 3 and message: _Send this message to 2 people via SMS._ should appear in step 2
- [ ] select back. editable textbox should appear with 3172956778, 2025468989 in it;
- [ ] add , 12345, 3172956778 to textbox; select continue; Message:  _We removed a total of 1 repeat number from your list._ should appear; Also message: _Of the 4 phone numbers you entered, we noticed problems with 1 of them. Please correct the phone number below and continue. Numbers must match the selected country's formatting rules._ should appear

**Comment by StaicyG** [comment]
@kinstelli @crcommons @Angamanga 

Passed QA test on .io

**Comment by Eve-wanderi** [comment]
Passed QA tests
